### PR TITLE
Relation.Binary.PropositionalEquality over-use,

### PR DESCRIPTION
**Pull request by JacquesCarette**
Many uses of `Relation.Binary.PropositionalEquality` (which is kind of 'fat' for such a low-level module) can usually be 
done instead as importing `.Core` …and/or `.Properties` instead.

### CODE REVIEW

File: src/Algebra/Solver/IdempotentCommutativeMonoid/Example.agda
Comment (MatthewDaggitt) [inline src/Algebra/Solver/IdempotentCommutativeMonoid/Example.agda:22]: Extra space here?

File: src/Data/List/Membership/DecPropositional.agda
Comment (MatthewDaggitt) [inline src/Data/List/Membership/DecPropositional.agda:13]: We definitely have a definition `DecidableEquality A` somewhere in the library....

File: src/Data/Product/Function/NonDependent/Propositional.agda
Comment (MatthewDaggitt) [inline src/Data/Product/Function/NonDependent/Propositional.agda:51]: This needs a CHANGELOG entry.

File: src/Function/Metric/Nat/Bundles.agda
Comment (MatthewDaggitt) [inline src/Function/Metric/Nat/Bundles.agda:25]: I feel this isn't a very useful narrowing of the import. `Bundles` is always going to want to import everything in `Structures`....

File: src/Function/Metric/Rational/Bundles.agda
Comment (MatthewDaggitt) [inline src/Function/Metric/Rational/Bundles.agda:24]: Same here.

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]
At this point, we should probably add a CHANGELOG entry as well, saying we've made an attempt to reduce the size of the dependency graph for many modules?

File: src/Data/List/Membership/DecPropositional.agda
Comment (Taneb) [reply src/Data/List/Membership/DecPropositional.agda:13]: It's in `Relation.Binary.Definitions`, which is conveniently already imported here

Comment (jamesmckinna) [reply src/Data/List/Membership/DecPropositional.agda:13]: NB. That definition clearly post-dates some parts of the library (Eg. `Axiom.UniquenessOfIdentityProofs`...) where the idiom `(_≟_ : Decidable (_≡_ {A = A}))` is still in force... so we should consider a separate issue/PR to tidy all of those up?

**Comment by jamesmckinna** [comment]
Monumental effort! I couldn't find much to comment on where @MatthewDaggitt had not already done so, so I'm not entirely confident about giving an "Approve" review, without *really* getting the fine-toothed comb out, but in principle, I think: merge!

**Comment by JacquesCarette** [comment]
Thanks @jamesmckinna . I am hoping to deal with the comments from Matthew in the next couple of days.

While quite time consuming, this exercise was also quite soothing and entertaining. Helped discover many things I'd like to discuss for a 3.0 version. As well as having me look at many parts of the library that I had no clue existed.

**Comment by jamesmckinna** [comment]
@JacquesCarette you wrote: 
> While quite time consuming, this exercise was also quite soothing and entertaining. Helped discover many things I'd like to discuss for a 3.0 version. As well as having me look at many parts of the library that I had no clue existed.

Perhaps, as momentum builds around v2.1/v3.0 issues/PRs, we should open a shopping list issue for v3.0, consolidating the discussions currently scattered through the repo?

File: src/Data/List/Membership/DecPropositional.agda
Comment (jamesmckinna) [reply src/Data/List/Membership/DecPropositional.agda:13]: I found only 5 or 6 such uses... so maybe it's not as bad as I thought!

Comment (jamesmckinna) [reply src/Data/List/Membership/DecPropositional.agda:13]: Er... make that more like 30! Oops...

Comment (jamesmckinna) [reply src/Data/List/Membership/DecPropositional.agda:13]: See #2354

Comment (JacquesCarette) [reply src/Data/List/Membership/DecPropositional.agda:13]: I'll wait until that PR gets merged, and then merge it in to this one.

File: src/Data/Product/Function/NonDependent/Propositional.agda
Comment (JacquesCarette) [reply src/Data/Product/Function/NonDependent/Propositional.agda:51]: Done, along with other addition to CHANGELOG

**Review by MatthewDaggitt — APPROVED** [review]
Looks great to me. Any chance you can resolve the merge conflicts and then we can hit merge!

**Pull request merged**
